src/chat/Chat.tsx

```diff
@@ -0,0 +1,186 @@
+import React, {useEffect, useState} from 'react';
+import 'react-chat-elements/dist/main.css';
+import './Chat.css';
+// @ts-ignore
+import { MessageList, Input, Button, ChatList } from 'react-chat-elements';
+import config from '../config.json';
+import {withRouter} from "react-router";
+import {connect} from "react-redux";
+import ChatModel from "./types/Chat";
+import Message from "./types/Message";
+
+const Chat = (props : any) => {
+    const [message, setMessage] = useState<string>('');
+    const [messageList, setMessageList] = useState();
+    const [chatList, setChatList] = useState()
+    const [chatEnabled, setChatEnabed] = useState<boolean>(false)
+
+    const loadChats = async () => {
+        const options: RequestInit = {
+            method: 'GET',
+            headers: {
+                'Content-Type': 'application/json'
+            },
+            mode: 'cors',
+            cache: 'default'
+        }
+        let result =  await fetch(config.SERVICES.COMMUNICATION_SERVICE + '/' + props.auth.User.id, options);
+        if (result.status !== 200) {
+            return;
+        }
+
+        let json : ChatModel[] = await result.json();
+        let list : any[] = []
+
+        for (let i =0; i < json.length; i++){
+            let model = json[i];
+            let chatUser = props.auth.User.id === model.buyer.id ? model.seller : model.buyer;
+
+            let chatView = {
+                id: model.id,
+                alt: chatUser.name,
+                title: chatUser.name,
+                subtitle: 'No messages yet',
+                date: new Date('1900-01-01'),
+                unread: model.unread
+            }
+            if (model.message !== null){
+                chatView.subtitle = model.message.text;
+                chatView.date = new Date(model.message.timeStamp)
+            }
+            list.push(chatView)
+        }
+        list.sort(function (a,b) {
+            if (a.date < b.date){
+                return 1;
+            } else return -1;
+        })
+
+        setChatList(list)
+    }
+
+    useEffect(() => {
+        loadChats();
+
+        // eslint-disable-next-line
+    }, [props.auth.User.id]);
```

Comment:
Waarom staat hier de id in van de gebruiker, waarom wil je dat die hierop de chat inlaad?